components/browser/session/src/main/java/mozilla/components/browser/session/engine/EngineObserver.kt

```diff
@@ -39,6 +39,10 @@ internal class EngineObserver(
     private val store: BrowserStore? = null
 ) : EngineSession.Observer {
 
+    override fun onNavigateBack() {
+        if (session.canGoBack) session.searchTerms = ""
```

Comment:
I think the `canGoBack()` check may need to happen in the engine implementation. Now we are calling `canGoBack` after calling `goBack()` and there's no guarantee whether this navigation has already happened or not..?

Reply:
You are right, thank you. I've moved the check in the engine implementations.